public/app/components/visualizations/PieChart.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import * as d3 from 'd3';
+
+
+class PieChart extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.renderChart = this.renderChart.bind(this);
+
+    this.state = {
+      width: 500,
+      height: 500,
+    };
+
+    this.pie = d3.pie()
+                .sort(null)
+                .value(d => d.value);
+
+    this.color = d3.scaleOrdinal(props.colors);
+  }
+
+
+  renderChart() {
+    const { width, height } = this.state;
+    const radius = Math.min(height, width) / 2;
+
+    const path = d3.arc()
+      .outerRadius(radius - 10)
+      .innerRadius(0);
+
+    const label = d3.arc()
+      .outerRadius(radius - 40)
+      .innerRadius(radius - 40);
+
+    const arc = this.gChart
+      .attr('transform', `translate(${width/2},${height/2})`)
+      .selectAll('.arc')
+      .data(this.pie(this.props.data))
+      .enter()
+      .append('g')
+      .attr('class', 'arc');
+
+    arc.append('path')
+      .attr('d', path)
+      .attr('fill', d => this.color(d.value));
+
+    const li = this.legend
+      .selectAll('li')
+      .data(this.props.data)
+      .enter()
+      .append('li')
+      .text(d => d.label);
+
+    li.append('span')
+      .style('background', d => this.color(d.value));
+  }
+
+
+  componentDidMount() {
+    this.chart = d3.select(`#${this.props.table}-pie`);
```

Comment:
You can use the React ref callbacks here too rather than d3 select.

Reply:
Using `d3.select()` wraps the element in a `Selection` object which is what supplies all of D3's methods for chaining.

Reply:
I tried using `ref` to pass into `d3.select()` but D3 wasn't having it.